packages/gasket-intl/src/components.js

```diff
@@ -0,0 +1,259 @@
+   * @returns {React.Component} wrapped component
+   */
+  return Component => {
+    /**
+     * Wrapper component which sets up providers and reducer hook
+     *
+     * @param {object} props - Component props
+     * @param {object} [props.pageProps] - Component props from a Next.js page
+     * @param {LocalesProps} [props.pageProps.localesProps] - Initial state from a Next.js page
+     * @returns {JSX.Element} element
+     */
+    function Wrapper(props = {}) {
+
+      // Support for wrapping Next.js App with data from get server side and static props
+      const { pageProps: { localesProps } = {} } = props; // eslint-disable-line react/prop-types
+
+      const [state, dispatch] = useReducer(helpers.reducer, localesProps || {}, helpers.init);
+
+      // If we have incoming pageProps, we need to update state but have to by
+      // mutation rather than issuing a dispatch to avoid re-renders and timing issues
+      if (localesProps) {
+        merge(state, localesProps);
+      }
+
+      const locale = localesProps?.locale || helpers.getActiveLocale();
+
+      const { status } = state;
+      const messages = (state.messages || {})[locale];
+      const contextValue = { locale, status, dispatch };
+
+      return (
+        <GasketIntlContext.Provider value={ contextValue }>
+          <IntlProvider locale={ locale } key={ locale } messages={ messages } initialNow={ Date.now() }>
+            <Component { ...props } />
+          </IntlProvider>
+        </GasketIntlContext.Provider>
+      );
+    }
+
+    Wrapper.displayName = `withIntlProvider(${ Component.displayName || Component.name || 'Component' })`;
+    if ('getInitialProps' in Component) {
+      Wrapper.getInitialProps = Component.getInitialProps;
+    }
+    return Wrapper;
+  };
+}
+
+/**
+ * Component that loads a locale file before rendering children
+ *
+ * @param {object} props - Props
+ * @param {LocalePathPart} props.localesPath - Path containing locale files
+ * @param {React.Component} [props.loading] - Custom component to show while loading
+ * @returns {JSX.Element|null} element
+ */
+export function LocaleRequired(props) {
+  const { localesPath, loading = null, children } = props;
+  const loadState = helpers.useGasketIntl(localesPath);
+  if (loadState === LOADING) return loading;
+  return <>{ children }</>;
+}
+
+LocaleRequired.propTypes = {
+  localesPath: PropTypes.string,
+  loading: PropTypes.node,
+  children: PropTypes.node.isRequired
+};
+
+LocaleRequired.defaultProps = {
+  localesPath: manifest.localesPath
+};
+
+/**
+ * Make an HOC that loads a locale file before rendering wrapped component
+ *
+ * @param {LocalePathPart} localesPath - Path containing locale files
+ * @param {object} [options] - Options
+ * @param {React.Component} [options.loading] - Custom component to show while loading
+ * @returns {function} wrapper
+ */
+export function withLocaleRequired(localesPath = manifest.localesPath, options = {}) {
```

Comment:
Maybe consider breaking apart this file and having something like

`src/components/index.js` - export from all the others
`src/components/with-locale-required.js` - just this hoc
etc.